pkg/services/libraryelements/database.go

```diff
@@ -441,6 +441,42 @@ func (l *LibraryElementService) getAllLibraryElements(c context.Context, signedI
 	if folderFilter.parseError != nil {
 		return model.LibraryElementSearchResult{}, folderFilter.parseError
 	}
+
+	var foldersWithMatchingTitles []string
+	if len(strings.TrimSpace(query.SearchString)) > 0 {
+		searchQuery := folder.SearchFoldersQuery{
+			OrgID:        signedInUser.GetOrgID(),
+			Title:        query.SearchString,
+			SignedInUser: signedInUser,
+		}
+
+		folderHits, err := l.folderService.SearchFolders(c, searchQuery)
+		if err != nil {
+			if strings.Contains(err.Error(), "cannot be called on the legacy folder service") {
```

Comment:
If we know in advance `folderService.SearchFolders` only works with `FlagKubernetesClientDashboardsFolders` we could maybe ask about it instead of using a string contains to match with an error string.